element_facemap/facemap_inference.py

```diff
@@ -0,0 +1,496 @@
+)
+
+schema = dj.schema()
+
+_linking_module = None
+
+
+def activate(
+    facemap_model_schema_name: str,
+    fbe_schema_name: str = None,
+    *,
+    create_schema: bool = True,
+    create_tables: bool = True,
+    linking_module: str = None,
+):
+    """Activate schema.
+
+    Args:
+        facemap_model_schema_name (str): Schema name on the database server to activate the
+            `facemap_pose_model` schema of element-facemap
+        fbe_schema_name (str):  Schema name on the database server to activate the 'facial_behavioral_estimation
+        create_schema (bool): When True (default), create schema in the database if it
+            does not yet exist.
+        create_tables (bool): When True (default), create tables in the database if
+            they do not yet exist.
+        linking_module (str): A module name or a module containing the required
+            dependencies to activate the `facial_behavior_estimation` module:
+
+    Dependencies:
+    Upstream tables:
+        + Session: A parent table to VideoRecording, identifying a recording session
+        + Equipment: A parent table to VideoRecording, identifying video recording equipment
+        + VideoRecording: A parent table to FacemapPoseEstimationTask, identifying videos to be used in inference
+    Functions:
+        + get_facemap_root_data_dir() -> list
+            Retrieves the root data directory(s) with face recordings for all
+            subject/sessions. Returns a string for the full path to the root data directory.
+        + get_facemap_processed_data_dir(session_key: dict) -> str
+            Optional function to retrieve the desired output directory
+            for Facemap files for a given session. If unspecified,
+            the output is stored in the video folder for the session, which is the default behavior of Facemap.
+            Returns a string of the absolute path of the output directory.
+    """
+    if isinstance(linking_module, str):
+        linking_module = importlib.import_module(linking_module)
+    assert inspect.ismodule(
+        linking_module
+    ), "The argument 'dependency' must be a module's name or a module"
+    assert hasattr(
+        linking_module, "get_facemap_root_data_dir"
+    ), "The linking module must specify a lookup function for a root data directory"
+
+    global _linking_module
+    _linking_module = linking_module
+
+    # activate facial behavioral extimation (fbe) schema
+    fbe.activate(
+        fbe_schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        linking_module=linking_module,
+    )
+
+    # activate facial pose model schema
+    schema.activate(
+        facemap_model_schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        add_objects=_linking_module.__dict__,
+    )
+
+
+# ----------------------------- Table declarations ----------------------
+@schema
+class BodyPart(dj.Lookup):
```

Comment:
Is this table better identified as `KeyPoints`?

Reply:
Yes, agreed. @sidhulyalkar would you rename to `KeyPoint`, here and everywhere else.
Facemap detects orofacial features, naming it body parts seems wrong.

Reply:
I agree but then I believe Facemap itself still refers to those as `bodyparts`, right?
The results are orofacial features, calling it body parts seems wrong. But if this is the terminology used in facemap, we should stick to the same term.

Reply:
Yes agreed, Facemap incorrectly uses the term 'BodyParts' to define their keypoints, but we should stick with their convention.